src/input/reader_to_storage.cc

```diff
@@ -140,9 +140,31 @@ StorageBase * ReaderToStorage::make_storage(PathBase &p, const Type::TypeBase *t
         return storage;
     }
 
-    // return Null storage if there is null on the current location
-    if (p.is_null_type())
-        return new StorageNull();
+    // create storage if Record is empty
+	if (typeid(*type) == typeid(Type::Record) && p.is_effectively_null()) {
+		return make_storage_empty_record(p, static_cast<const Type::Record *>(type));
```

Comment:
No need for specific make_storage for empty record. Just check that path.get_record_key_set returns true and set empty set for null path.